src/libUtils/ProcessStats.cpp

```diff
@@ -105,4 +119,44 @@ int64_t DisplayPhysicalMemoryStats(const string& str, int64_t startMem) {
     LOG_GENERAL(INFO, "PM diff = " << diff << " " << str);
   }
   return processPhysMemUsedMB;
-}
\ No newline at end of file
+}
```

Comment:
I'm a little suspicious about in-process collection of memory and cpu stats about the process itself. 

Why isn't it done with some form of probes that are then exported to a centralized metrics collection. We had discussed OpenTelemetry etc, - those solutions already have probes set up at hosts, that can collect this information from all nodes, aggregated any way w/o having to write and maintain this code.